File: 2316-count-hills-and-valleys-in-an-array/2316-count-hills-and-valleys-in-an-array.c
int countHillValley (int* nums, int numsSize) {
    int count = 0;
    bool diff [2] = {0, 0};
    for (int i = 1; i < numsSize; i++){
        if (nums [i - 1] == nums [i]) continue;
        bool bigger = nums [i] > nums [i - 1];
        diff [bigger] = 1;
        count += diff [bigger] && diff [!bigger];
        diff [!bigger] = 0;
    }
    return count;
}

File: 2316-count-hills-and-valleys-in-an-array/2316-count-hills-and-valleys-in-an-array.java
class Solution {
    public int countHillValley (int [] nums) {
        int count = 0, prev = nums [0];
        int diff [] = new int [2], i = 0;
        while (i < nums.length) {
            while (i < nums.length && prev == nums [i]) i += 1;
            if (i == nums.length) break;
            int bigger;
            if (nums [i] > prev) bigger = 1;
            else bigger = 0;
            diff [bigger] = 1;
            count += ((diff [bigger] == 1) && (diff [1 - bigger] == 1)) ? 1 : 0;
            diff [1 - bigger] = 0;
            prev = nums [i];
            i++;
        }
        return count;
    }
}